### PR TITLE
Testing enhancements

### PR DESCRIPTION
**Pull request by levibyte**

### CODE REVIEW

**Comment by levibyte** [comment]
Long list of enhancements in test infrastructure side.

File: src/core/application.hpp [hunk 1]
```diff
@@ -10,32 +10,39 @@ enum appMode {APPNORM, APPREPLAY, APPTESTING, APPLOAD};
...
-	}
-	/**/
-	
-
-	static bool is_replay_mode() { return (Application::getInstance().get_mode() == APPREPLAY); }
-	static bool is_testing_mode() { return (Application::getInstance().get_mode() == APPTESTING); }
-	static bool is_load_mode() { return (Application::getInstance().get_mode() == APPLOAD); }
+    appMode m_mode;
+    public:
+        bool m_is_debug = false;
```
Comment (nagaina) [inline src/core/application.hpp]: public?

File: src/core/application.hpp [hunk 2]
```diff
@@ -10,32 +10,39 @@ enum appMode {APPNORM, APPREPLAY, APPTESTING, APPLOAD};
...
+            m_is_debug = QString::fromLocal8Bit(qgetenv("ELEN_PAINTER_TESTDBG").constData()).isEmpty()?false:true;
+        }
+        
+        void set_replay_mode(bool m) { 
+            Messenger::expose_msg(info,"Set to mode"+QString::number(int(m)).toStdString());
+            m_mode = m?APPREPLAY:APPNORM; 
+        }
+        
+        void set_mode(const appMode& m) { m_mode = m; } 
+        appMode get_mode() { return m_mode; }
```
Comment (nagaina) [inline src/core/application.hpp]: get_mode() const ...

File: src/gui/console.hpp [hunk 3]
```diff
@@ -26,12 +28,13 @@ class ConsoleWidget : public QFrame
...
 private slots:
 	void onCommandEntered();
+        void onConsoleLinkClicked(QUrl);
 
 private:
 	virtual bool eventFilter(QObject*, QEvent*) override;
 
 private:
-	QTextEdit* m_view = nullptr;
+	QTextBrowser* m_view = nullptr;
```
Comment (nagaina) [inline src/gui/console.hpp]: why we need textbrowser?

**Pull request merged**